### PR TITLE
Feat watch history

### LINKED ISSUE #17: Todo: Add Watch History Feature

### PR DESCRIPTION
**Pull request by coldpigli**
1) Videos added to history when played
2) Individual videos can be removed from history
3) The entire history can be cleaned.

### CODE REVIEW

**Review by tanishq20 — APPROVED** [review]
Very clean code. Easy to understand. Like the way you used hook functions.

File: src/pages/History/History.jsx [hunk 1]
```diff
@@ -1,9 +1,51 @@
-import React from 'react'
+import { HorizontalVideoCard, RedirectToLogin, TopNav } from "../../components";
+import { useUserDetails } from "../../contexts";
+import { checkLogin } from "../../utils";
+import { clearHistory, removeFromHistory } from "../../utils/handleHistory";
+import styles from "./History.module.css";
```
Comment (srejitk) [inline src/pages/History/History.jsx:5]: You can try using an index file and cleaning the imports to make it neater later on.

File: src/utils/handleHistory.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            authorization: localStorage.getItem(tokenName),
+          },
+        });
+        if (res.status === 200 || res.status === 201) {
+          const { history } = res.data;
+          toast({ type: "success", message: "Removed from History" });
+          dispatchUser({type: "REMOVE_FROM_HISTORY", payload: history})   
+        }
+      } catch (err) {
+        console.log("oops something bad happed", err);
```
Comment (srejitk) [inline src/utils/handleHistory.js:41]: You can remove this log since there's a Toast anyway

File: src/utils/handleHistory.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            authorization: localStorage.getItem(tokenName),
+          },
+        });
+        if (res.status === 200 || res.status === 201) {
+          const { history } = res.data;
+          toast({ type: "success", message: "Cleared your History" });
+          dispatchUser({type: "CLEAR_HISTORY", payload: history})   
+        }
+      } catch (err) {
+        console.log("oops something bad happed", err);
```
Comment (srejitk) [inline src/utils/handleHistory.js:61]: Same suggestion here

**Review by srejitk — APPROVED** [review]
Code looks neat. Just clean up the imports later on. Good work.

**Review by rdshinde — APPROVED** [review]
I liked the modular approach. Seems good to merge. 👍

**Pull request merged**